In [ ]:
#Load revelant packages
import numpy as np
import pandas as pd
import statsmodels.api as sm


In [ ]:
from google.colab import files

In [ ]:
from google.colab import files
uploaded = files.upload()

Saving rfj_data.csv to rfj_data (1).csv


In [ ]:
#simple linear regression of Tropicana
y = df[['q1']]
X = df[['p1']]
X = sm.add_constant(X)
model = sm.OLS(y, X).fit()
est = sm.OLS(y, X).fit()
print(est.summary())


                            OLS Regression Results                            
Dep. Variable:                     q1   R-squared:                       0.564
Model:                            OLS   Adj. R-squared:                  0.562
Method:                 Least Squares   F-statistic:                     279.4
Date:                Sat, 15 Nov 2025   Prob (F-statistic):           8.43e-41
Time:                        20:33:08   Log-Likelihood:                -3389.0
No. Observations:                 218   AIC:                             6782.
Df Residuals:                     216   BIC:                             6789.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const       1.691e+07   7.82e+05     21.631      0.0

In [ ]:
#simple linear regression of Minute Maid
y = df[['q2']]
X = df[['p2']]
X = sm.add_constant(X)
model = sm.OLS(y, X).fit()
est = sm.OLS(y, X).fit()
print(est.summary())

                            OLS Regression Results                            
Dep. Variable:                     q2   R-squared:                       0.403
Model:                            OLS   Adj. R-squared:                  0.400
Method:                 Least Squares   F-statistic:                     145.7
Date:                Sat, 15 Nov 2025   Prob (F-statistic):           5.62e-26
Time:                        20:36:31   Log-Likelihood:                -3432.2
No. Observations:                 218   AIC:                             6868.
Df Residuals:                     216   BIC:                             6875.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const       1.217e+07   8.11e+05     15.006      0.0

In [ ]:
#simple linear regression of Private Label
y = df[['q3']]
X = df[['p3']]
X = sm.add_constant(X)
model = sm.OLS(y, X).fit()
est = sm.OLS(y, X).fit()
print(est.summary())

                            OLS Regression Results                            
Dep. Variable:                     q3   R-squared:                       0.367
Model:                            OLS   Adj. R-squared:                  0.364
Method:                 Least Squares   F-statistic:                     125.0
Date:                Sat, 15 Nov 2025   Prob (F-statistic):           3.38e-23
Time:                        20:37:23   Log-Likelihood:                -3441.5
No. Observations:                 218   AIC:                             6887.
Df Residuals:                     216   BIC:                             6894.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const       1.218e+07   8.57e+05     14.216      0.0

In [ ]:
print(df.describe())


             WEEK            q1            q2            q3          p1  \
count  218.000000  2.180000e+02  2.180000e+02  2.180000e+02  218.000000   
mean   109.500000  3.935819e+06  2.477531e+06  2.691890e+06    0.039134   
std     63.075352  2.072420e+06  2.159166e+06  2.187939e+06    0.004695   
min      1.000000  2.944640e+05  8.584660e+05  7.536000e+05    0.024588   
25%     55.250000  2.538064e+06  1.296548e+06  1.447744e+06    0.035979   
50%    109.500000  3.309412e+06  1.597826e+06  1.994592e+06    0.039729   
75%    163.750000  4.772334e+06  2.808464e+06  3.034720e+06    0.042173   
max    218.000000  1.790061e+07  1.695846e+07  1.318880e+07    0.048420   

               p2          p3  
count  218.000000  218.000000  
mean     0.032512    0.024242  
std      0.004596    0.003384  
min      0.016464    0.013935  
25%      0.031025    0.022431  
50%      0.032608    0.024134  
75%      0.035373    0.026079  
max      0.040248    0.035574  


Average Price for Tropicana = 0.0391 ($/oz)

Average Price for Minute Maid = 0.0325 ($/oz)

Average Price for Private Label = 0.0242 ($/oz)



---



Average Quantity for Tropicana = 3.94 (million/oz)

Average Quantity for Minute Maid = 2.48 (million/oz)

Average Quantity for Private Label = 2.69 (million/oz)

Price Elasticity = average price/average quantity* slope


---
Price Elasticity for Tropicana = (0.0391/3.94)*-331,500,000= -3.29

Price Elasticity for Minute Maid = (0.0325/2.48)*-298,200,000= -3.91

Price Elasticity for Private Label = (0.0242/2.69)* -391,500,000= -3.52

All three brands have elastic demand (|ε| > 1), and especially Minute Maid and Private Label are very sensitive to price at the sample means.


Optimal Price = (slope * 0.01-intercept)/2 * slope

Optical Price for Tropicana = (-331,500,000 * 0.01-16,910,000)/2 * -331,500,000= 0.0305

Optical Price for Minute Maid = (-298,200,000 * 0.01-12,170,000)/2 * -298,200,000=0.0254

Optimal Price for Private label = (-391,500,000 * 0.01-12,180,000)/2 * -391,500,000=0.0206


At the current average prices, all three brands operate in the elastic portion of their demand curves (elasticities between -3.3 and -3.9). This implies that a small price increase would reduce revenue, while small price cuts would increase revenue.

The simple monopoly pricing model with MC = $0.01/oz suggests that all three brands’ optimal prices are lower than their historical average prices. Lowering prices from current levels toward these optimal points would increase predicted quantities sold quite substantially.

Among the three, Minute Maid appears to have the most elastic demand (highest |ε| at the means), indicating its customers are especially price-sensitive, while Tropicana and Private Label are slightly less so (though still very elastic).
